dagshub/models/model_locator.py

```diff
@@ -0,0 +1,320 @@
+                log_message(f"Loading the model from bucket {model_path}", logger=logger)
+                return BucketModelLoader(self.repo_api, PurePosixPath(model_path))
+            except PathNotFoundError:
+                raise ModelNotFoundError
+        else:
+            try:
+                self.repo_api.list_path(model_path, revision=self.git_ref)
+                log_message(f"Loading the model from repo {model_path}", logger=logger)
+                return RepoModelLoader(self.repo_api, self.git_ref, PurePosixPath(model_path))
+            except PathNotFoundError:
+                raise ModelNotFoundError
+
+    lookup_dirs = ["model", "models"]
+
+    def try_find_model_in_repo(self) -> Optional[ModelLoader]:
+        for lookup_dir in self.lookup_dirs:
+            modelPath = PurePosixPath(lookup_dir)
+            try:
+                self.repo_api.list_path(str(modelPath), revision=self.git_ref)
+                log_message(f"Loading the model from repo directory {modelPath}", logger=logger)
+                return RepoModelLoader(repo_api=self.repo_api, revision=self.git_ref, path=modelPath)
+            except PathNotFoundError:
+                continue
+        return None
+
+    def try_find_model_in_bucket(self, bucket: str) -> Optional[ModelLoader]:
+        bucketPath: Optional[PurePosixPath] = None
+        for storage in self.repo_storages:
+            if storage.name.startswith(bucket):
+                bucketPath = PurePosixPath(f"{storage.protocol}/{storage.name}")
+                break
+
+        if bucketPath is None:
+            return None
+
+        for lookup_dir in self.lookup_dirs:
+            modelPath = bucketPath / lookup_dir
+            try:
+                self.repo_api.list_storage_path(str(modelPath))
+                log_message(f"Loading the model from bucket directory {modelPath}", logger=logger)
+                return BucketModelLoader(repo_api=self.repo_api, path=modelPath)
+            except PathNotFoundError:
+                continue
+
+        return None
+
+    def try_find_model_in_dagshub_storage(self) -> Optional[ModelLoader]:
+        bucketPath = PurePosixPath("s3") / self.repo_api.repo_name
+        for modelPath in self.lookup_dirs:
+            try:
+                self.repo_api.list_storage_path(str(bucketPath / modelPath))
+                log_message(f"Loading the model from DagsHub Storage directory {modelPath}", logger=logger)
+                return DagsHubStorageModelLoader(repo_api=self.repo_api, path=PurePosixPath(modelPath))
+            except PathNotFoundError:
+                continue
+
+        return None
+
+    def find_model(self) -> ModelLoader:
+        model_loader: Optional[ModelLoader]
+
+        def return_if_exists_throw_otherwise():
+            if model_loader is None:
+                raise ModelNotFoundError
+            if model_loader is not None:
+                return model_loader
```

Comment:
I don't really see the point in that function you only use it in one return point, with a simple if-else.

Reply:
Ah, I used it in two places, but later removed one of those (the one with the specified path).
Inlined it, thanks